apiproxy/resources/jsc/set-response.js

```diff
@@ -12,40 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
- //prepare response object
- 
- var jws = {
-    token: context.getVariable('jwtmessage')
- };
- 
- if (context.getVariable('grant_type') === 'client_credentials' || context.getVariable('grant_type') === 'password') {
-    if (context.getVariable("scp")) {
-     jws.scope = context.getVariable("scp");
-    }
-    
-    jws.access_token = context.getVariable('jwtmessage');
-    jws.token_type   = "Bearer";
-
-     // for /token flow
-     jws.expires_in = context.getVariable("oauthv2accesstoken.AccessTokenRequest.expires_in");
-
-     // for any other flows if any 
-    if ( !jws.expires_in ) {
-      jws.expires_in   = parseInt( context.getVariable("token_expiry"), 10) / 1000; // convert to seconds
-    }
-    
-    //if refresh token exists, add it to response
-    if (context.getVariable('grant_type') === "password") {
-        jws.refresh_token            = context.getVariable("oauthv2accesstoken.AccessTokenRequest.refresh_token");
-        jws.refresh_token_expires_in = context.getVariable("oauthv2accesstoken.AccessTokenRequest.refresh_token_expires_in");         
-        jws.refresh_token_issued_at  = context.getVariable("oauthv2accesstoken.AccessTokenRequest.refresh_token_issued_at") ; 
-        jws.refresh_token_status     = context.getVariable("oauthv2accesstoken.AccessTokenRequest.refresh_token_status"); 
-    }
- }
- 
- 
- //send response
- context.setVariable("response.header.Content-Type","application/json");
- context.setVariable("response.header.Cache-Control","no-store");
- context.setVariable("response.header.Pragma","no-cache");
- context.setVariable("response.content", JSON.stringify(jws));
+/* global context */
+
+var payload = {
+      token: context.getVariable('jwtmessage')
+    };
+
+if (context.getVariable('grant_type') === 'client_credentials' || context.getVariable('grant_type') === 'password') {
+  if (context.getVariable("scp")) {
+    payload.scope = context.getVariable("scp");
+  }
+
+  payload.access_token = context.getVariable('jwtmessage');
+  payload.token_type   = "Bearer";
+
+  // for /token flow
+  payload.expires_in = parseInt(context.getVariable("oauthv2accesstoken.AccessTokenRequest.expires_in"), 10);
+
+  // for any other flows if any
+  if ( !payload.expires_in ) {
+    payload.expires_in   = parseInt( context.getVariable("token_expiry"), 10) / 1000; // convert to seconds
+  }

```

Comment:
If context.getVariable returns null (which is handled below with the check), the parseint will throw an exception.
It needs to be handled differently.

The correct fix has already been submitted as a part of the next release